src/components/Modal/index.js

```diff
@@ -0,0 +1,140 @@
+import React, { useState, useEffect, useRef } from 'react'
+
+import { Dialog } from '@material-ui/core'
+import { ThemeProvider } from '@material-ui/core/styles'
+import Icon from '~/components/Icon'
+import PropTypes from 'prop-types'
+import Img from 'gatsby-image'
+import {
+  theme,
+  useStylesMainModal,
+  DialogContentContainer,
+  ModalCloseIncon,
+  ModalContent,
+  ModalTitleContainer,
+  ModalTitleContent,
+  ModalTextContent,
+  ModalLinksFooter,
+  ModalTitleFooter,
+  ModalDialogContainer,
+} from './styles'
+import Title from '../Utilities/Title'
+import Text from '../Utilities/Text'
+
+const Modal = ({ open, setOpen, query }) => {
+  const classes = useStylesMainModal()
+  const [linkState, setLinkState] = useState(query.linkIndex)
+
+  const linksArray = query.links
+  const currentLink = linksArray[linkState]
+  const nextLink = linksArray[linkState + 1]
+
+  const ministriesArray = query.ministries
+  const ministriesContent = []
+
+  ministriesArray
+    ? ministriesArray.map((info) => {
+        if (currentLink === info.list_title.text) {
+          ministriesContent.push(info)
+        }
+      })
+    : ''
+
+  const isFirstRun = useRef(true)
+  useEffect(() => {
+    if (isFirstRun.current) {
+      isFirstRun.current = false
+      return
+    }
+    setLinkState(query.linkIndex)
+  }, [query.linkIndex])
+
+  const handleClickNext = () => {
+    let i = linkState
+    if (i >= 0 && i < linksArray.length - 1) {
+      setLinkState(linkState + 1)
+    }
+  }
+
+  const handleClickPrev = () => {
+    let i = linkState
+    if (i > 0) {
+      setLinkState(linkState - 1)
```

Comment:
this will send user to last slide if on first slide and clicks prev

`
else if (i === 0) {
 setLinkState(linksArray.length - 1)
}
`

Reply:
same principal can be apply for next